### PR TITLE
Fix alert bug

### PR DESCRIPTION
**Pull request by nunnatsa**
Fix Prometheus alert bug, where the KubevirtHyperconvergedClusterOperatorInstallationNotCompletedAlert alert is not working when the HyperConverged CR is not present.

This alert is all about the existence of the HyperConverged CR, and so when it removed with the CR, the alert is not working. The root cause is that HCO manages the PrometheusRule resource as any other resource, and set its owner resource to the HyperConverged CR. In addition, the code that reconciles the PrometheusRule resource is only run when the HypeConverged CR is found. 

This PR moves the PrometheusRule handling code to a new package name `alerts`. Then the new PrometheusRule reconcile function is called regardless the existence of the HyperConverged CR. Also, the PrometheusRule ownerReference is now set to the hco-operator Deployment, so it will be garbage collected on the operator uninstallation. 

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
Fix Prometheus alert bug, where the KubevirtHyperconvergedClusterOperatorInstallationNotCompletedAlert alert is not working when the HyperConverged CR is not present.
```

### CODE REVIEW

File: controllers/alerts/alerts.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+		Labels: map[string]string{
+			severityAlertLabelKey:  "info",
+			partOfAlertLabelKey:    partOfAlertLabelValue,
+			componentAlertLabelKey: componentAlertLabelValue,
+		},
+	}
+}
+
+// Recording rules for openshift/cluster-monitoring-operator
+func createRequestCPUCoresRule() monitoringv1.Rule {
```
Comment (tiraboschi) [inline controllers/alerts/alerts.go:231]: is this needed?

File: controllers/hyperconverged/hyperconverged_controller_test.go [hunk 2]
```diff
@@ -714,7 +714,7 @@ var _ = Describe("HyperconvergedController", func() {
 				).To(BeNil())
 
 				Expect(foundResource.Status.RelatedObjects).ToNot(BeNil())
-				Expect(len(foundResource.Status.RelatedObjects)).Should(Equal(18))
+				Expect(len(foundResource.Status.RelatedObjects)).Should(Equal(17))
```
Comment (tiraboschi) [inline controllers/hyperconverged/hyperconverged_controller_test.go:717]: is this due to the prometheus Rule referencing the deployment?
maybe we should still list it under relatedObjects

File: controllers/alerts/alerts.go (hunk 1 above)
Comment (nunnatsa) [reply controllers/alerts/alerts.go:231]: yes. it was there before the PR. Just wrapped it in a function

**Comment by tiraboschi** [comment]
ignoring slightly decreased coverage
/override coverage/coveralls

**Comment by nunnatsa** [comment]
hco-e2e-upgrade-index-aws lane passed
/override ci/prow/hco-e2e-upgrade-index-azure

File: controllers/hyperconverged/hyperconverged_controller_test.go (hunk 2 above)
Comment (nunnatsa) [reply controllers/hyperconverged/hyperconverged_controller_test.go:785]: Fixed. I added the PrometheusRule to the relatedObjects list, but the implementation is in the `alerts` package, so it won't affect this test.

**Review by tiraboschi — APPROVED** [review]

**Pull request merged**